app/inflation/README.md

```diff
@@ -0,0 +1,97 @@
+
+# UMEE Tokenomic Restructuring Proposal
+
+## Introduction
+
+The UMEE Tokenomic Restructuring proposal aims to optimize the tokenomics of the $UMEE token, which is issued following the standard Cosmos SDK mechanism. The goal is to progressively limit the token issuance in an algorithmic way, taking into account various factors such as network security, adoption of liquid staking, and additional income sources for stakers through interchain security and chain-based revenue streams.
+
+## Design Overview
+
+1. The $UMEE 2.0 Tokenomics will adopt a similar Cosmos dynamic inflation mechanism based on the bonding rate of the network. This allows the Umee chain to naturally adjust the inflation rate to drive staking activities and reach the targeted bonding rate.
+
+2. To initiate the new tokenomics, the maximum and minimum inflation rates will decrease by 25%. Subsequent decreases will occur every two years.
+
+   - Current min and max inflation rates: 7% - 14%
+   - New max and min inflation rates at the start: 5.6% - 11.2%
+
+3. The max and min inflation rates will decrease by 25% every two years in inflation cycles. This is in anticipation of the growth of the Umee ecosystem and wider adoption of liquid staking.
+
+   - Each inflation cycle will have a variable yearly total amount of newly emitted tokens.
+   - At the end of each inflation cycle, the `min_inflation_rate`, `max_inflation_rate` will be decreased by 25%.
+   - The inflation rate change speed will be accelerated from 1 year to 6 months.
+
+4. The total $UMEE supply will be capped at 21 billion tokens.
+
+   - Inflation will be reduced during each inflation cycle until minting stops when the maximum supply is reached.
+   - Minting will resume if the supply drops below 21 billion, and it will stop again once the 21 billion mark is reached.
+   - The supply will be `StakingTokenSupply`.
+
+5. All inflationary tokens will be distributed to stakers and delegators, similar to the current mechanism.
+
+6. **Umee governance will have the power to vote and modify the `inflation params`` if the external environment significantly changes, necessitating a different approach.**
+
+## Technical Specification
+
+The Inflation Params to be stored in the `x/ugov` module:
+
+```go
+// max supply based on the bank/QueryTotalSupplyRequest 
+max_supply       sdk.Coin
+// length of the inflation cycle as described in the Design
+inflation_cycle_duration  time.Duration
+// in basis points. 12 corresponds to 0.012. New inflation is calculated as:
+// old_inflation * (10'000 - inflation_reduction_rate)/10'000
+inflation_reduction_rate sdk.Dec
+```
+
+See [Ugov Proto](https://github.com/umee-network/umee/blob/main/proto/umee/ugov/v1/ugov.proto) for inflation params
+
+### InflationCalculationFn
+
+The inflation calculation logic will be implemented as follows:
+
+```bash
```

Comment:
Is there a reason this is presented as a bash script?

It would be easier to read as a list of steps, or maybe as go pseudocode if comments and variable names were kept succinct